templates/definition/meter/sonnenbatterie_eco56.yaml

```diff
@@ -0,0 +1,70 @@
+template: sonnenbatterie-eco56
+products:
+  - brand: Sonnenbatterie
+    description:
+      generic: comfort, eco 5, eco 6, oem 6.5
+params:
+  - name: usage
+    choice: ["grid", "pv", "battery"]
+  - name: host
+  - name: port
+    default: 7979
+  - name: capacity
+    advanced: true
+render: |
+  type: custom
+  {{- if eq .usage "grid" }}
+  power:
+    source: calc
+    add:
+    - source: http
+      uri: http://{{ .host }}:{{ .port }}/rest/devices/battery
+      jq: .M39 # current purchase at the interconnection point
+    - source: http
+      uri: http://{{ .host }}:{{ .port }}/rest/devices/battery
+      jq: .M38 # current feed-in at the interconnection point
+      scale: -1
+  energy:
+    source: http
+    uri: http://{{ .host }}:{{ .port }}/rest/devices/battery
+    jq: .M41 # cumulated purchase since installation
+  {{- end }}
+  {{- if eq .usage "pv" }}
+  power:
+    source: http
+    uri: http://{{ .host }}:{{ .port }}/rest/devices/battery
+    jq: .M03 # current pv power
+  energy:
+    source: http
+    uri: http://{{ .host }}:{{ .port }}/rest/devices/battery
+    jq: .M37 # cumulated pv production since installation of Sonnenbatterie
+  {{- end }}
+  {{- if eq .usage "battery" }}
+  power:
+    source: calc
+    add:
+    # The API is bugged and returns 65525 instead of 0 in some cases
+    - source: http
+      uri: http://{{ .host }}:{{ .port }}/rest/devices/battery
+      # M35 current charging power
+      # S65 max inverter power
+      jq: "if .M35 <= .S65 then .M35 else 0 end"
+      scale: -1
+    - source: http
+      uri: http://{{ .host }}:{{ .port }}/rest/devices/battery
+      # M34 current discharging power
+      # S65 max inverter power
+      jq: "if .M34 <= .S65 then .M34 else 0 end"
+  energy:
+    source: http
+    uri: http://{{ .host }}:{{ .port }}/rest/devices/battery
+    jq: .M31 # total stored energy over lifetime
+  soc:
+    source: http
+    uri: http://{{ .host }}:{{ .port }}/rest/devices/battery
+    jq: .M30 # SOC relative to usable capacity
+    #jq: .M05 # display SOC
+  {{- if .capacity }}
+  capacity: {{ .capacity }} # kWh
```

Comment:
capacity is just a fix value from config. It can not take "dynamic" values.

Its only used for multi-battery setups with different capacities of each battery to calculate the total soc (weight of each soc).

Reply:
My proposal would be to change that to be able to get that info from the system. Just one less thing to configure (where possible of course). Even though it´s only nice to have.
If there is interest in this I could open a PR.

Reply:
Lets do that if we can fine enough systems that would profit.

Reply:
Bei den Speichern können das viele selbst irgendwo mitteilen, um nicht zu sagen fast alle.

Reply:
Hier aber OT ;)